### ISSUE TITLE
Problems when using a custom KubeJS crafting recipe in an RFTools crafter

### ISSUE THREAD

**Issue by itsjusttriz** [McJty/RFTools#2164]
**Issue description:**
Beta testing a modpack named Project Isothermal: Expert, and we have a couple custom items and their recipes made with KubeJS.
While playing, I tried to place 2 of these custom recipes on the Tier 1 Crafter. 1 of these recipes work, whereas the other doesnt. I assumed the 2nd wasnt working due to the 1st using 1 of the same ingredients, so i removed the 1st recipe from the Crafters memory. Still no result.

Working Recipe is a dirt block made by a shapeless 2x2 of Mud or something. Mud is an item made in KubeJS.
Problematic Recipe is a shapeless 2x2 requiring x2 of the before mentioned, Mud, and x2 of another custom item (*cant remember the name of*) in the shape of:
```
XO
OX
```

**Steps to reproduce:** 
Create temporary custom items that are mentioned above, and create their recipes using KubeJS. Attempt crafts in a Tier 1 crafter.

**Versions:**
* Minecraft: 1.16.5
* Forge: 36.2.34
* CompatLayer (only if on Minecraft 1.10 or 1.11):
* McJtyLib: 1.16-5.1.4
* RFTools: https://gyazo.com/790b4034eb5dd633baa850d1616710b5

**Possibly incompatible mods and versions:** 
Using Optifine.

Relevant logs, if any:

**Comment by McJty** [comment]
Are these custom recipes that KubeJS adds standard recipes? i.e. can you make them in a regular crafting table?

**Comment by itsjusttriz** [comment]
> Are these custom recipes that KubeJS adds standard recipes? i.e. can you make them in a regular crafting table?

Yes they are. Basically the first 2 crafts required for this modpack

**Comment by McJty** [comment]
Hmm not sure then. I have no clue what KubeJS does. The crafter simply looks at the standard recipes returned by the recipe manager. I think you'll have to ask the KubeJS guys if they have a clue what could be wrong.

**Comment by itsjusttriz** [comment]
Aight thank you.

**Comment by itsjusttriz** [comment]
@McJty Turns out i was mistaken. The recipe ingredients were custom items made by KubeJS, but the recipe itself was put together with CraftTweaker. The dev of the pack im playing also had the same issue with his previous modpack. Recipe made by CT and attempted automation in the Crafter, with no result. Maybe something to look into?

**Comment by Kikiisyourfriend** [comment]
Aye. So the items in question are custom items made through KubeJS but the craft itself is from CT.
In my other pack Damage Inc I had a custom CT craft for Minecraft Sticks with Minecraft Planks yielding 2 sticks instead of 4 and they were not being auto crafted in the Crafter.
Now what's strange is not all CT crafts act this way. I have some going right now that are created with CT and they are working fine.
A bit strange.
For example my Flesh Block craft using CT in Project Isothermal Expert modpack crafts fine in the Crafter.

**Comment by McJty** [comment]
Not much I can do about this I'm afraid. As I said the rftools crafter simply uses the standard recipe manager. I have no clue what I could do differently

**Comment by Kikiisyourfriend** [comment]
I appreciate your time. 😺

**Comment by jaredlll08** [comment]
The CraftTweaker Shapeless recipe is a normal `ICraftingRecipe` that works fine in all other crafters (AE2 / RefinedStorage / Mekanism), so not sure what we could be doing wrong.

<https://github.com/CraftTweaker/CraftTweaker/blob/e8cd826790b94e1eeff34b1ccab22778b603c8ca/src/main/java/com/blamejared/crafttweaker/impl/recipes/CTRecipeShapeless.java#L28>
